packages/teleterm/src/ui/appContext.ts

```diff
@@ -86,10 +96,32 @@ export default class AppContext implements IAppContext {
       this.workspacesService,
       this.clustersService
     );
+
+    this.reloginService = new ReloginService(
+      mainProcessClient,
+      this.modalsService,
+      this.clustersService
+    );
+    this.tshdNotificationsService = new TshdNotificationsService(
+      this.notificationsService,
+      this.clustersService
+    );
   }
 
   async init(): Promise<void> {
+    this.setUpTshdEventSubscriptions();
     await this.clustersService.syncRootClusters();
     this.workspacesService.restorePersistedState();
   }
+
+  private setUpTshdEventSubscriptions() {
+    this.subscribeToTshdEvent('relogin', ({ request, onCancelled }) => {
+      // The handler for the relogin event should return only after the relogin procedure finishes.
+      return this.reloginService.relogin(request, onCancelled);
+    });
+
+    this.subscribeToTshdEvent('sendNotification', ({ request }) => {
```

Comment:
Are these `notifications` related only to `relogin` events? Or they can occur in other cases? 
I'm wondering if we could combine `ReloginService` and `TshdNotificationsService`.

Reply:
Currently those notifications are tied to the relogin process but that's only because so far it's the only place where we send a notification from tshd to the Electron app. In the future we could add different subjects to `SendNotificationRequest` and handle different notifications through a single RPC.